src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ManifestResourceBlockingPolicy.cs

```diff
@@ -96,7 +96,12 @@ public AssemblyFeatureInfo(EcmaModule module, Logger logger, IReadOnlyDictionary
                             ms = new UnmanagedMemoryStream(reader.CurrentPointer, length);
                         }
 
-                        BlockedResources = SubstitutionsReader.GetSubstitutions(logger, module.Context, ms, resource, module, "resource " + resourceName + " in " + module.ToString(), featureSwitchValues);
+                        var featureSwitches = new Dictionary<string, bool>(featureSwitchValues);
+
+                        // Considered always true
+                        featureSwitches["System.Resources.UseSystemResourceKeys"] = true;
```

Comment:
Need to inline knowledge about this here so that we remove the manifest resources. We either inline them, or use resource keys. Either way they're not needed. Don't like having to hardcode this feature switch value.

Reply:
I think we should have a different feature switch for this. The substitutions should allow for the same resource to be removed "Twice" via two different feature switches - it's not perfect, but should work.
But this goes back to the discussion of how much should the compiler know about the implementation.